pkg/apis/garden/validation/validation_test.go

```diff
@@ -245,11 +245,66 @@ var _ = Describe("validation", func() {
 
 					errorList := ValidateCloudProfile(awsCloudProfile)
 
-					Expect(len(errorList)).To(Equal(1))
+					Expect(len(errorList)).To(Equal(3))
 					Expect(*errorList[0]).To(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeInvalid),
+						"Field": Equal(fmt.Sprintf("spec.%s.constraints.machineImages", fldPath)),
+					}))
+					Expect(*errorList[1]).To(MatchFields(IgnoreExtras, Fields{
```

Comment:
Use the [`ConsistOf` matcher](https://github.com/gardener/gardener/blob/master/pkg/apis/garden/validation/validation_test.go#L1700) for all new tests and all tests that you adapted.